### PR TITLE
statistics: build the table analysis job queue

### PR DESCRIPTION
**Pull request by hi-rustin**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tidb/issues/50132

Problem Summary:

#### Description

This pull request completes the process for analyzing non-partitioned tables in the job queue. 

It calculates three metrics: 
1) change percent
2) table size (cols * row_count)
3) detection of any special events (new index)

There is also a dummy calculator added, I will implement it after this PR is done.

#### Explanation of Changes

- Added support for building analysis job queue. (only for non-partitioned tables)
- Also added tests for it.
- Moved some common functions to the exec package.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by hi-rustin — COMMENTED** [review]
🔢 Self-check

File: pkg/statistics/handle/autoanalyze/priorityqueue/calculator.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
+// Copyright 2023 PingCAP, Inc.
```
Comment (hawkingrei) [inline pkg/statistics/handle/autoanalyze/priorityqueue/calculator.go:1]:
```suggestion
// Copyright 2024 PingCAP, Inc.
```

File: pkg/statistics/handle/autoanalyze/refresher/refresher.go [hunk 2]
```diff
@@ -87,3 +99,192 @@ func (r *Refresher) pickOneTableAndAnalyzeByPriority() {
...
+					if _, ok := lockedTables[tbl.Meta().ID]; ok {
+						continue
+					}
+
+					tblInfo := tbl.Meta()
+					if tblInfo.IsView() {
+						continue
+					}
+					pi := tblInfo.GetPartitionInfo()
+					pushJob := func(job *priorityqueue.TableAnalysisJob) {
```
[the remark this answers is not in the document]
Comment (hi-rustin) [reply pkg/statistics/handle/autoanalyze/refresher/refresher.go:146]: The reason I haven't made it as a function of refresher is that we need to get the latest auto-analyze ratio every time. Therefore, we need to create a calculator here.

[the remark this answers is not in the document]
Comment (hi-rustin) [reply pkg/statistics/handle/autoanalyze/refresher/refresher.go]: Renamed.

File: pkg/statistics/handle/autoanalyze/priorityqueue/calculator.go (hunk 1 above)
Comment (hi-rustin) [reply pkg/statistics/handle/autoanalyze/priorityqueue/calculator.go]: Updated.

**Review by hawkingrei — APPROVED** [review]

**Review by hi-rustin — COMMENTED** [review]
🔢 Self-check

**Review by hi-rustin — COMMENTED** [review]
🔢 Self-check

**Review by qw4990 — APPROVED** [review]

**Pull request merged**